src/type-layout.md

```diff
@@ -280,19 +280,29 @@ The `align` representation can be used on `struct`s and `union`s to raise the
 alignment of the type to a given value.
 
 Alignment is specified as a parameter in the form of `#[repr(align(x))]`. The
-alignment value must be a power of two of type `u32`. The `align` representation
-can raise the alignment of a type to be greater than it's primitive alignment,
-it cannot lower the alignment of a type.
+alignment value must be a power of two up to 2<sup>29</sup>. The `align`
+representation can raise the alignment of a type to be greater than it's
+primitive alignment, it cannot lower the alignment of a type.
 
 The `align` and `packed` representations cannot be applied on the same type and
 a `packed` type cannot transitively contain another `align`ed type.
 
 ### The `packed` Representation
 
-The `packed` representation can only be used on `struct`s and `union`s.
+The `packed` representation can be used on `struct`s and `union`s to lower the
+alignment and padding of the type.
+
+The packing value is specified as a parameter in the form of
+`#[repr(packed(x))]`. If no value is given, as in `#[repr(packed)]`, then the
+packing value is 1. The packing value must be a power of two up to
+2<sup>29</sup>.
 
-It modifies the representation (either the default or `C`) by removing any
-padding bytes and forcing the alignment of the type to `1`.
+The `packed` representation sets the alignment to the smaller of the specified
+packing and the current alignment (either default or `C`). The alignments of
```

Comment:
Here the "default" alignment is referenced but above "primitive alignment" is used... we should pick one phrasing.

Reply:
Get rid of both "current' and "primitive". Get rid of the parenthetical. Instead spell it out explicitly that if the type has the default or C representation, and the alignment from that representation is smaller, it will use that alignment.